// ng-knife/esm2015/lib/directives/only-numbers.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, HostListener } from '@angular/core';
export class OnlyNumbersDirective {
    constructor() {
        this.onlyNumbersRegExp = new RegExp('^\\d+$');
    }
    /**
     * @param {?} e
     * @return {?}
     */
    pasteClipboard(e) {
        const /** @type {?} */ data = e.clipboardData.getData('text');
        return this.onlyNumbersRegExp.test(data) ? true : false;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    onKeyDown(e) {
        return (e.ctrlKey || e.metaKey || e.keyCode === 8 || e.keyCode === 9 || e.keyCode === 37 || e.keyCode === 39 ||
            e.keyCode === 46 || e.keyCode === 91 || e.keyCode === 93) ? true : this.onlyNumbersRegExp.test(e.key);
    }
}
OnlyNumbersDirective.decorators = [
    { type: Directive, args: [{
                selector: '[knifeOnlyNumbers]'
            },] },
];
OnlyNumbersDirective.propDecorators = {
    pasteClipboard: [{ type: HostListener, args: ['paste', ['$event'],] }],
    onKeyDown: [{ type: HostListener, args: ['keydown', ['$event'],] }]
};
function OnlyNumbersDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    OnlyNumbersDirective.prototype.onlyNumbersRegExp;
}
