// ng-knife/bundles/ng-knife.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('ng-knife', ['exports', '@angular/core'], factory) :
    (factory((global['ng-knife'] = {}),global.ng.core));
}(this, (function (exports,core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NoSpacesDirective = (function () {
        function NoSpacesDirective() {
        }
        /**
         * @param {?} e
         * @return {?}
         */
        NoSpacesDirective.prototype.onKeyDown = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                return !(e.keyCode === 32);
            };
        NoSpacesDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[knifeNoSpaces]'
                    },] },
        ];
        NoSpacesDirective.propDecorators = {
            onKeyDown: [{ type: core.HostListener, args: ['keydown', ['$event'],] }]
        };
        return NoSpacesDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NoCopyPasteDirective = (function () {
        function NoCopyPasteDirective() {
        }
        /**
         * @param {?} e
         * @return {?}
         */
        NoCopyPasteDirective.prototype.noCopy = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                return false;
            };
        /**
         * @param {?} e
         * @return {?}
         */
        NoCopyPasteDirective.prototype.noPaste = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                return false;
            };
        /**
         * @param {?} e
         * @return {?}
         */
        NoCopyPasteDirective.prototype.noCut = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                return false;
            };
        NoCopyPasteDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[knifeNoCopyPaste]'
                    },] },
        ];
        NoCopyPasteDirective.propDecorators = {
            noCopy: [{ type: core.HostListener, args: ['copy', ['$event'],] }],
            noPaste: [{ type: core.HostListener, args: ['paste', ['$event'],] }],
            noCut: [{ type: core.HostListener, args: ['cut', ['$event'],] }]
        };
        return NoCopyPasteDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var AlphanumericDirective = (function () {
        function AlphanumericDirective() {
        }
        /**
         * @param {?} e
         * @return {?}
         */
        AlphanumericDirective.prototype.onKeyDown = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                return (e.keyCode === 8 || e.keyCode === 32 || e.keyCode === 37 || e.keyCode === 39)
                    ? true : (new RegExp('^[a-zA-Z0-9_]*$')).test(e.key);
            };
        AlphanumericDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[knifeAlphanumeric]'
                    },] },
        ];
        AlphanumericDirective.propDecorators = {
            onKeyDown: [{ type: core.HostListener, args: ['keydown', ['$event'],] }]
        };
        return AlphanumericDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var OnlyNumbersDirective = (function () {
        function OnlyNumbersDirective() {
            this.onlyNumbersRegExp = new RegExp('^\\d+$');
        }
        /**
         * @param {?} e
         * @return {?}
         */
        OnlyNumbersDirective.prototype.pasteClipboard = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                var /** @type {?} */ data = e.clipboardData.getData('text');
                return this.onlyNumbersRegExp.test(data) ? true : false;
            };
        /**
         * @param {?} e
         * @return {?}
         */
        OnlyNumbersDirective.prototype.onKeyDown = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                return (e.ctrlKey || e.metaKey || e.keyCode === 8 || e.keyCode === 9 || e.keyCode === 37 || e.keyCode === 39 ||
                    e.keyCode === 46 || e.keyCode === 91 || e.keyCode === 93) ? true : this.onlyNumbersRegExp.test(e.key);
            };
        OnlyNumbersDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[knifeOnlyNumbers]'
                    },] },
        ];
        OnlyNumbersDirective.propDecorators = {
            pasteClipboard: [{ type: core.HostListener, args: ['paste', ['$event'],] }],
            onKeyDown: [{ type: core.HostListener, args: ['keydown', ['$event'],] }]
        };
        return OnlyNumbersDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgKnifeModule = (function () {
        function NgKnifeModule() {
        }
        NgKnifeModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [],
                        declarations: [
                            NoSpacesDirective,
                            NoCopyPasteDirective,
                            AlphanumericDirective,
                            OnlyNumbersDirective
                        ],
                        exports: [
                            NoSpacesDirective,
                            NoCopyPasteDirective,
                            AlphanumericDirective,
                            OnlyNumbersDirective
                        ]
                    },] },
        ];
        return NgKnifeModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.NgKnifeModule = NgKnifeModule;
    exports.ɵc = AlphanumericDirective;
    exports.ɵb = NoCopyPasteDirective;
    exports.ɵa = NoSpacesDirective;
    exports.ɵd = OnlyNumbersDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
