// ng-knife/fesm2015/ng-knife.js
import { Directive, HostListener, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NoSpacesDirective {
    /**
     * @param {?} e
     * @return {?}
     */
    onKeyDown(e) {
        return !(e.keyCode === 32);
    }
}
NoSpacesDirective.decorators = [
    { type: Directive, args: [{
                selector: '[knifeNoSpaces]'
            },] },
];
NoSpacesDirective.propDecorators = {
    onKeyDown: [{ type: HostListener, args: ['keydown', ['$event'],] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NoCopyPasteDirective {
    /**
     * @param {?} e
     * @return {?}
     */
    noCopy(e) {
        return false;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    noPaste(e) {
        return false;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    noCut(e) {
        return false;
    }
}
NoCopyPasteDirective.decorators = [
    { type: Directive, args: [{
                selector: '[knifeNoCopyPaste]'
            },] },
];
NoCopyPasteDirective.propDecorators = {
    noCopy: [{ type: HostListener, args: ['copy', ['$event'],] }],
    noPaste: [{ type: HostListener, args: ['paste', ['$event'],] }],
    noCut: [{ type: HostListener, args: ['cut', ['$event'],] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class AlphanumericDirective {
    /**
     * @param {?} e
     * @return {?}
     */
    onKeyDown(e) {
        return (e.keyCode === 8 || e.keyCode === 32 || e.keyCode === 37 || e.keyCode === 39)
            ? true : (new RegExp('^[a-zA-Z0-9_]*$')).test(e.key);
    }
}
AlphanumericDirective.decorators = [
    { type: Directive, args: [{
                selector: '[knifeAlphanumeric]'
            },] },
];
AlphanumericDirective.propDecorators = {
    onKeyDown: [{ type: HostListener, args: ['keydown', ['$event'],] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class OnlyNumbersDirective {
    constructor() {
        this.onlyNumbersRegExp = new RegExp('^\\d+$');
    }
    /**
     * @param {?} e
     * @return {?}
     */
    pasteClipboard(e) {
        const /** @type {?} */ data = e.clipboardData.getData('text');
        return this.onlyNumbersRegExp.test(data) ? true : false;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    onKeyDown(e) {
        return (e.ctrlKey || e.metaKey || e.keyCode === 8 || e.keyCode === 9 || e.keyCode === 37 || e.keyCode === 39 ||
            e.keyCode === 46 || e.keyCode === 91 || e.keyCode === 93) ? true : this.onlyNumbersRegExp.test(e.key);
    }
}
OnlyNumbersDirective.decorators = [
    { type: Directive, args: [{
                selector: '[knifeOnlyNumbers]'
            },] },
];
OnlyNumbersDirective.propDecorators = {
    pasteClipboard: [{ type: HostListener, args: ['paste', ['$event'],] }],
    onKeyDown: [{ type: HostListener, args: ['keydown', ['$event'],] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgKnifeModule {
}
NgKnifeModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [
                    NoSpacesDirective,
                    NoCopyPasteDirective,
                    AlphanumericDirective,
                    OnlyNumbersDirective
                ],
                exports: [
                    NoSpacesDirective,
                    NoCopyPasteDirective,
                    AlphanumericDirective,
                    OnlyNumbersDirective
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgKnifeModule, AlphanumericDirective as ɵc, NoCopyPasteDirective as ɵb, NoSpacesDirective as ɵa, OnlyNumbersDirective as ɵd };
